src/cursor.c

```diff
@@ -421,47 +421,36 @@ cursor_motion_absolute(struct wl_listener *listener, void *data)
 	process_cursor_motion(seat->server, event->time_msec);
 }
 
-static void
-handle_cursor_button_with_meta_key(struct view *view, uint32_t button,
-		double lx, double ly)
-{
-	if (!view) {
-		return;
-	}
-
-	/* move */
-	if (button == BTN_LEFT) {
-		interactive_begin(view, LAB_INPUT_STATE_MOVE, 0);
-		return;
-	}
-
-	/* resize */
-	uint32_t edges;
-	edges = lx < view->x + view->w / 2 ? WLR_EDGE_LEFT : WLR_EDGE_RIGHT;
-	edges |= ly < view->y + view->h / 2 ? WLR_EDGE_TOP : WLR_EDGE_BOTTOM;
-	interactive_begin(view, LAB_INPUT_STATE_RESIZE, edges);
-}
-
-static void
+static bool
 handle_release_mousebinding(struct server *server, uint32_t button, uint32_t modifiers, enum ssd_part_type view_area, uint32_t resize_edges)
 {
 	struct mousebind *mousebind;
+	bool activated_any = false;
+	bool activated_any_frame = false;
+
 	wl_list_for_each_reverse(mousebind, &rc.mousebinds, link) {
-		if (mousebind->context == view_area
+		if (ssd_part_contains(mousebind->context, view_area)
 				&& mousebind->button == button
 				&& modifiers == mousebind->modifiers) {
-			if (mousebind->mouse_event
-					== MOUSE_ACTION_RELEASE) {
-				action(server, mousebind->action,
-					mousebind->command, resize_edges);
-			}
-			if (mousebind->pressed_in_context) {
-				mousebind->pressed_in_context = false;
-				action(server, mousebind->action,
-					mousebind->command, resize_edges);
+			switch (mousebind->mouse_event) {
+				case MOUSE_ACTION_RELEASE:
+					break;
+				case MOUSE_ACTION_CLICK:
+					if (mousebind->pressed_in_context) {
+						mousebind->pressed_in_context = false;
+						break;
+					}
+					continue;
+				default:
+					continue;
```

Comment:
This logic requires that `wl_list_for_each_reverse` supports `continue` like a native loop.